src/lib/runtime_config/runtime_config.ml

```diff
@@ -47,25 +47,98 @@ module Json_layout = struct
       module Timed = struct
         type t =
           { initial_minimum_balance: Currency.Balance.t
-          ; cliff_time: Coda_numbers.Global_slot.t (*slot number*)
-          ; vesting_period: Coda_numbers.Global_slot.t (*slots*)
+          ; cliff_time: Coda_numbers.Global_slot.t
+          ; vesting_period: Coda_numbers.Global_slot.t
           ; vesting_increment: Currency.Amount.t }
-        [@@deriving yojson, dhall_type]
+        [@@deriving yojson, dhall_type, sexp]
+      end
+
+      module Permissions = struct
+        module Auth_required = struct
+          type t = None | Either | Proof | Signature | Both | Impossible
+          [@@deriving yojson, dhall_type, sexp, bin_io_unversioned]
+        end
+
+        type t =
+          { stake: bool
+          ; edit_state: Auth_required.t
+          ; send: Auth_required.t
+          ; receive: Auth_required.t
+          ; set_delegate: Auth_required.t
+          ; set_permissions: Auth_required.t
+          ; set_verification_key: Auth_required.t }
+        [@@deriving yojson, dhall_type, sexp, bin_io_unversioned]
+      end
+
+      module Token_permissions = struct
+        type disable_new_accounts = {disable_new_accounts: bool}
+        [@@deriving yojson, dhall_type, sexp, bin_io_unversioned]
+
+        type account_disabled = {account_disabled: bool}
+        [@@deriving yojson, dhall_type, sexp, bin_io_unversioned]
+
+        type t =
+          | Token_owned of disable_new_accounts
+          | Not_owned of account_disabled
+        [@@deriving yojson, dhall_type, sexp, bin_io_unversioned]
+      end
+
+      module Snapp_account = struct
+        (*TODO: representation for the snapp account type*)
```

Comment:
This TODO and the one below should be done. The type is effectively

```ocaml
{state: Field.t list; verification_key: string}
```

The state list has fixed length 8 (anything else should be an error) and the verification key can just be the binary encoding of a `Pickles.Side_loaded_verification_key.t`, which should be handled in `Genesis_ledger_helper`.

Once this is done, can you move `Snapp_lib`'s contents back into `Coda_base`? I agree that it should be broken up, but the natural divisions to my mind are `Coda_account`, `Coda_ledger` and `Coda_transactions`, and nothing else seems to benefit from its being isolated.